### DISCUSSION TITLE
Powerwall Capacity Graph

### DISCUSSION THREAD

**Discussion by jasonacox** [jasonacox/Powerwall-Dashboard#34] [Announcements]
A suggestion came in from a reddit thread to add Powerwall Charge metrics and a graph to monitor the degradation of the Powerwall capacity over time.   I have updated the project to now include this for the `dashboard-animation.json` dashboard.  It would be easy to port this to the other dashboards or your custom dashboard by copying the panel JSON. 

* The graph shows the `PW1_POD_nom_full_pack_energy` which represents the estimated capacity of the Powerwall. It will be interesting to see how that trends down over time due to battery degradation. 
* The graph also shows the `PW1_POD_nom_energy_remaining` which represents the amount of "charge" the Powerwall is holding so we will see this climb during charging cycles and fall during discharge. 
* I set the panel to display PW1 through PW6.  If you have more Powerwalls, you will need to edit the panel to add them.

<img width="1076" alt="image" src="https://user-images.githubusercontent.com/836718/174494485-f901cb79-09ae-4674-88a5-7af00e89fb89.png">

To upgrade, you will need to upgrade the telegraf config and add a downsampling job to influxdb.  The `upgrade.sh` script will do most of this for you but older upgrade.sh did not include a telegraf upgrade so follow the steps below.  It's always a good idea to backup your setup before changing. 

```bash
# remove existing telegraf config
docker stop telegraf
docker rm telegraf

# upgrade setup
bash upgrade.sh
```

**Comment by cwagz** [comment]
This is a great idea. I really like the additions to the Power Flow dashboard. Overlaying home usage on the 12-month solar generation chart is also really nice.

**Comment by jasonacox** [comment]
Update on capacity graph.  If you isolate the PWs over 3+ days (hint: click on PW1 in the legend), you can see the gradual decline in capacity.  It's both very cool and very depressing. :) 

<img width="1248" alt="image" src="https://user-images.githubusercontent.com/836718/173506530-98239509-c22f-466d-88f6-d0e37cb264db.png">

**Reply by jasonacox** [reply]
Update - isolating to PW1

<img width="1195" alt="image" src="https://user-images.githubusercontent.com/836718/178628084-5fd6c82f-cef5-4440-9862-4f8450bbfcdc.png">


The step-down and step back up is an interesting behavior.

**Reply by youzer-name** [reply]
I'm not seeing quite the same thing on my Powerwalls. Here is my chart after I set the Y axis to auto and decimal places to 3:

![image](https://user-images.githubusercontent.com/57046429/178731654-600c2f6a-e169-482b-b50f-633f21437880.png)

While the lowest point is over toward the right, the most recent point is about in the middle of the values for the last 30 days. 

For my 2nd battery, it is actually going up:

![image](https://user-images.githubusercontent.com/57046429/178732090-677e14de-fbe3-4ab2-a444-531be18eaffa.png)

Here is both: 

![image](https://user-images.githubusercontent.com/57046429/178731913-b78a28e8-ea3e-4bc9-968c-d45e570eb0b4.png)


I've read that with Tesla car batteries the range displayed depends on the BMS calculation of total battery capacity, and that its calculations can be influenced by the recent charge/discharge cycles. Cycling the pack down to low SOC and back up to 100% can result in the reported range increasing. (That is based on what I've read online, so take it for what it's worth).

Here is my 30-day charge level for PW1. (PW2 looks pretty much identical). I keep a 30% reserve level and I've hit that a few times in the last 30 days during cloudy/rainy weather.

![image](https://user-images.githubusercontent.com/57046429/178732940-841598dd-b72b-4362-b729-db08b359c490.png)

Here is what that looks like as a percentage. I think the zero on 6/28 was a data dropout during a firmware upgrade.

![image](https://user-images.githubusercontent.com/57046429/178734856-96d64e12-741c-4370-b0a0-10ff896259bb.png)

**Reply by jasonacox** [reply]
Wow, that's interesting @youzer-name - I'm clearly seeing degradation, but I'm also cycling from 20% to 100% back to 20% nearly every day due to A/C.  I wonder if I raise my reserve to 30% if it would help or moving to Time-Based-Control.

<img width="1133" alt="image" src="https://user-images.githubusercontent.com/836718/178894746-f61c24c8-50f6-486c-82df-68d4cbd67abd.png">
<img width="1129" alt="image" src="https://user-images.githubusercontent.com/836718/178894813-b5073d59-6f1d-42ab-9186-718920f64c56.png">

**Reply by mcbirse** [reply]
Interesting graphs! This prompted me to check my system and how it compares.

I am also seeing degradation over the past 30 days. Like @jasonacox this is cycling from 20% to near 100% and back to 20% each day, due to A/C use for heating (it has been very cold in Sydney, with severe weather recently (major flooding in fact), and not much sun at times where the Powerwall hardly charged - which you can see in the graphs).

![image](https://user-images.githubusercontent.com/108725631/179186077-4456098e-16c8-4fa5-bcdb-612e4f3f2c7e.png)
![image](https://user-images.githubusercontent.com/108725631/179186099-6ce6f842-d10f-4bb9-9943-822f33f76d4b.png)

@youzer-name - I'm curious to know why your Powerwalls show max capacity of almost 15kWh? Are they new, or a different model to my Powerwall 2 battery?

I'm not sure what my capacity was when new (installed Apr last year)... would have been great to have this data logged from then to see how much it degrades over a year?

**Reply by youzer-name** [reply]
@mcbirse My Powerwalls were installed on February 28, 2022, so they are a bit newer than yours. They are Powerwall 2 (not the Plus).

**Reply by jasonacox** [reply]
Latest graph of my two Powerwalls (Powerwall Capacity) operating in _self-powered_ mode.  I raised my reserve to 30% (first strike line) and thought it had been the cause of the leveling off but switched back to 20% on 9/1 and haven't seen a lot of change.  I want to add a graph line on here for "expected" capacity degradation based on the 70% at 10 years warranty. 

<img width="699" alt="image" src="https://user-images.githubusercontent.com/836718/189578625-91808764-74b0-4196-903b-fdf90f574e5a.png">
